### PR TITLE
feature/473 update wpgraphql for posts2posts to include input arguments Natac update

### PR DESCRIPTION
**Pull request by natac13**
- p2p where input arguments & p2p mutation fields added
- users query where input args fixed
- graphql folder added
- styling changes
- chore: move things to better dir structure
- attempt
- dev: add phpcs and phpstan to dev dependencies.
- .
- dev: fix plugin prefixes in phpcs
- dev: format and lint by wpcs
- fix: include autoloader in repo
- dev: remove wp-unit tests and enforce php7.4
- chore: phpstan level 8
- chore: update minimum WPGraphQL Version
- chore: add readme.txt
- dev: refactor PR
- fix: include autoload in repo
- fix: dont access post_types directly.
- fix: return early if `$query_args['postToPostConnections']` is not set
- feat: use get_graphql_register_action()
- fix: only capture_p2p_connections once.
- chore: remove extra line
- fix: update muation connection string to enum

### CODE REVIEW

File: src/Mutations/PostMutation.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+				}
+
+				$connected_type = $post_to_post_connection['connection'];
+
+				if ( ( ! $post_to_post_connection['append'] || 0 === count( $post_to_post_connection['ids'] ) ) && false === strpos( $mutation_name, 'Create' ) ) {
+					$connection_to = null;
+
+					foreach ( $connections as $connection ) {
+						if ( $connection['name'] === $post_to_post_connection['connection'] ) {
+							$connection_to = $connection['to'];
```
Comment (justlevine) [inline src/Mutations/PostMutation.php:70]: Should there be a `break` here?

Comment (natac13) [reply src/Mutations/PostMutation.php:70]: Not sure what you mean?

File: src/Traits/ObjectsTrait.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace WPGraphQLPostsToPosts\Traits;
+
+use P2P_Connection_Type;
+
+trait ObjectsTrait {
```
Comment (justlevine) [inline src/Traits/ObjectsTrait.php:7]: This trait should be renamed to `PostTypes` or something else relevant.

File: src/Types/Post.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+			$connection_name = $post_type->name;
+
+			$connections = array_filter( $p2p_connections_to_map, fn( $p2p_connection ) => $p2p_connection['from'] === $connection_name || $p2p_connection['to'] === $connection_name );
+
+			foreach ( $connections as $connection ) {
+				array_push( $field_names, $connection['name'] );
+			}
+		}
+
+		if ( ! isset( $query_args['postToPostConnections'] ) ) {
```
Comment (justlevine) [inline src/Types/Post.php:53]: Under what circumstances would `postToPostConnections` not be set? Returning early might not be the correct approach.

File: src/Types/User.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+			$connections = array_filter( $p2p_connections_to_map, fn( $p2p_connection ) => $p2p_connection['from'] === $connection_name || $p2p_connection['to'] === $connection_name );
+
+			foreach ( $connections as $connection ) {
+				array_push( $field_names, $connection['name'] );
+			}
+		}
+
+		if ( ! isset( $query_args['postToPostConnections'] ) ) {
+			return $query_args;
```
Comment (justlevine) [inline src/Types/User.php:48]: When is `postToPostConnections` not set? Should we be throwing an error or returning early?

File: src/Mutations/PostMutation.php (hunk 1 above)
Comment (justlevine) [reply src/Mutations/PostMutation.php:70]: It's a foreach loop for the sole purpose of setting a value under an if condition. Are we iterating through the entire array to set it on the last value, or break the foreach on the first match?

**Pull request merged**